extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -66,20 +62,89 @@ void build(RecorderContext recorder,
+    }
+
+    private void registerCompulsoryClasses(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, StreamsPartitionAssignor.class));
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, DefaultPartitionGrouper.class));
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, DefaultProductionExceptionHandler.class));
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, FailOnInvalidTimestamp.class));
+    }
+
+    private void registerClassesThatClientMaySpecify(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        registerExceptionHandler(reflectiveClasses);
+        registerDefaultSerdes(reflectiveClasses);
+    }
+
+    private void registerExceptionHandler(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        Properties properties = buildKafkaStreamsProperties();
+        String exceptionHandlerClassName = properties
+                .getProperty(StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG);
+
+        if (exceptionHandlerClassName == null)
+            registerDefaultExceptionHandler(reflectiveClasses);
+        else
+            registerClassName(reflectiveClasses, exceptionHandlerClassName);
+    }
+
+    private void registerDefaultExceptionHandler(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, LogAndFailExceptionHandler.class));
+    }
+
+    private void registerDefaultSerdes(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+        Properties properties = buildKafkaStreamsProperties();
+        String defaultKeySerdeClass = properties.getProperty(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG);
+        String defaultValueSerdeClass = properties.getProperty(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG);
+
+        if (defaultKeySerdeClass != null)
+            registerClassName(reflectiveClasses, defaultKeySerdeClass);
+        if (defaultValueSerdeClass != null)
+            registerClassName(reflectiveClasses, defaultValueSerdeClass);
+        if (allDefaultSerdesAreNotDefinedInProperties(defaultKeySerdeClass, defaultValueSerdeClass)) {
+            registerDefaultSerde(reflectiveClasses);
+        }
+    }
+
+    private void registerClassName(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses, String defaultKeySerdeClass) {
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, defaultKeySerdeClass));
+    }
+
+    private boolean allDefaultSerdesAreNotDefinedInProperties(String defaultKeySerdeClass, String defaultValueSerdeClass) {
+        return defaultKeySerdeClass == null || defaultValueSerdeClass == null;
+    }
+
+    private void registerDefaultSerde(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
```

Comment:
Perhaps have a more generic `register()` method to be used everywhere:

    register(reflectiveClasses, ByteArraySerde.class);

Reply:
Sure!